### PR TITLE
Backends/cruxmir

### PR DESCRIPTION
**Pull request by fshaked**
Initial support for crux-mir.

This will break things with cargo-verify. In particular, when running cargo in compatibility-test you must specify the verifier like this: cargo test --features "verify propverify/verifier-crux"

### CODE REVIEW

File: verification-annotations/src/verifier_std.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+// Reject the current execution path with a verification success.
+// This is equivalent to assume(false)
+// and the opposite of report_error.
+//
+// Typical usage is in generating symbolic values when the value
+// does not meet some criteria.
+pub fn reject() -> ! {
+    // TODO: ???
+    panic!("reject")
```
Comment (alastairreid) [inline verification-annotations/src/verifier_std.rs:45]: I don't think that there is any reasonable way to implement this
so I worry that this will cause problems with strategies like 'char' that use this function.

File: verification-annotations/src/crux.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+// except according to those terms.
+
+/////////////////////////////////////////////////////////////////
+// FFI wrapper for Crux-mir static simulator tool
+/////////////////////////////////////////////////////////////////
+
+#[cfg(crux)]
+extern crate crucible;
+
+// pub mod crucible::symbolic;
```
Comment (alastairreid) [inline verification-annotations/src/crux.rs:16]: Delete this commented out code?

File: propverify/src/strategy.rs [hunk 3]
```diff
@@ -153,29 +153,55 @@ pub fn prop_is_replay() -> bool {
...
+        #[cfg_attr(crux,crux_test)]
+        // $(#[$meta])*
+        fn $test_name() {
+            $(
+                let str = stringify!($meta);
+                if str.starts_with("should_panic") {
+                    verifier::expect_raw(str);
+                }
+            )*
+            eprintln!("____META END____");
```
Comment (alastairreid) [inline propverify/src/strategy.rs:174]: Can we change the string from "____META END____" to something that starts with "VERIFIER"?
Maybe "VERIFIER_META_END"?

File: verification-annotations/src/verifier_std.rs [hunk 4]
```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 The Propverify authors
```
Comment (alastairreid) [inline verification-annotations/src/verifier_std.rs:1]: If I understand this correctly, this crate exists only to allow us to let us print the meta-annotations to support "should_panic".
In particular, I think it only exists to avoid linking errors?

So could we change every function body to just "panic()"?

File: compatibility-test/src/main.rs [hunk 5]
```diff
@@ -16,9 +16,9 @@
 
 #![allow(unused_imports)]
 
-#[cfg(not(verify))]
+#[cfg(not(feature = "verify"))]
```
Comment (alastairreid) [inline compatibility-test/src/main.rs:19]: Is this change needed for Crux?

**Review by alastairreid — CHANGES_REQUESTED** [review]
Can you
- drop the --cfg=verify -> --features="verify" change - unless it is needed by crux?
- change the ____META END___ string
- in verifier_std.rs, do one of
  - eliminate verifier_std.rs completely - it is not clear whether this is needed at all when in 'meta' mode
  - merge verifier_std.rs and verifier_crux.rs using conditional compilation
  - replace all function bodies with "panic()"
- squash changes that conflict with each other
   (It is hard to review a CL that makes a change in one commit and then changes it again in a later commit)

**Comment by alastairreid** [comment]
Has all the code in this change been committed in other changes now?
If so, should we delete this pull request?

**Comment by fshaked** [comment]
A better implementation was merged.
